lib/cocoapods/installer/xcode/pods_project_generator/file_references_installer.rb

```diff
@@ -28,6 +38,16 @@ def initialize(sandbox, pod_targets, pods_project)
             @sandbox = sandbox
             @pod_targets = pod_targets
             @pods_project = pods_project
+            @keep_folder_structure = false
+            
+            PODFILE_NAMES.each do |filename|
+              candidate = (Pathname sandbox.root).parent + filename
+              if candidate.exist?
+                if candidate.read.include? 'keep_folder_structure'
```

Comment:
this definitely not the way functionality would be added to the Podfile - if you want to introduce new features there then most likely the Podfile DSL is where to do it. See here: https://github.com/CocoaPods/Core/blob/master/lib/cocoapods-core/podfile/dsl.rb